ttnn/cpp/ttnn/operations/pool/maxpool/device/kernels/dataflow/reader_max_pool_2d_multi_core_sharded_with_halo_large_kernel_v2.cpp

```diff
@@ -104,53 +102,46 @@ void kernel_main() {
 
     uint32_t in_w_padded = in_w + 2 * pad_w;
 
-    uint32_t npages_to_reserve = nblocks;
-    uint32_t num_8_tile_blocks = 1;
+    uint32_t npages_to_reserve = 1;
     uint32_t read_bytes = in_nbytes_c;
     if (in_nbytes_c > MAX_ELE_PER_REDUCTION) {
-        num_8_tile_blocks = in_nbytes_c / MAX_ELE_PER_REDUCTION;
         read_bytes = MAX_ELE_PER_REDUCTION;  // for now, pow of 2 channels are only supported.
     }
     uint32_t counter = reader_id;
     uint32_t total_elems_to_reduce = window_h * window_w;
     uint32_t remaining_elems = total_elems_to_reduce % max_rows_for_reduction;
     while (counter < reader_nindices) {
-        for (uint32_t j = 0; j < num_8_tile_blocks; j++) {
-            for (uint32_t i = 0; i < nblocks; ++i) {
-                uint16_t top_left_local_index = reader_indices_ptr[counter];
-                uint32_t h_multiples = 0;
-                uint32_t processed_rows = 0;
-                uint32_t out_l1_write_addr_base = get_write_ptr(in_cb_id);
-                uint32_t out_l1_write_addr = out_l1_write_addr_base;
-                cb_reserve_back(in_cb_id, npages_to_reserve);
-                // If next is last chunk, fill whole buffer with -inf.
-                if ((total_elems_to_reduce - processed_rows) < max_rows_for_reduction)
-                    fill_with_val(out_l1_write_addr, TILE_SIZE * MAX_TILES_PER_REDUCTION, minus_inf);
-                for (uint32_t h = 0; h < window_h; ++h, h_multiples += in_w_padded) {
-                    uint32_t stick_offset = top_left_local_index + h_multiples;
+        for (uint32_t c_i = 0; c_i < in_nblocks_c; c_i++) {
+            uint16_t top_left_local_index = reader_indices_ptr[counter];
+            uint32_t processed_rows = 0;
+            cb_reserve_back(in_cb_id, npages_to_reserve);
+            uint32_t out_l1_write_addr_base = get_write_ptr(in_cb_id);
+            uint32_t out_l1_write_addr = out_l1_write_addr_base;
+            if ((total_elems_to_reduce - processed_rows) < max_rows_for_reduction)
```

Comment:
Shall we keep this comment??
`// If next is last chunk, fill whole buffer with -inf.`

Reply:
It's still there down lower, but I have added a pertinent comment here as well!